numpy/lib/nanfunctions.py

```diff
@@ -991,7 +995,10 @@ def _nanpercentile1d(arr1d, q, overwrite_input=False, interpolation='linear'):
     s = np.where(c)[0]
     if s.size == arr1d.size:
         warnings.warn("All-NaN slice encountered", RuntimeWarning)
-        return np.nan
+        if q.ndim == 0:
+            return np.nan
+        else:
+            return np.nan * np.ones((len(q),))
```

Comment:
Right, thought we had added that, but must have been some other thing (maybe it was just how q is interpreted if 1-d).